metrics/density/fast_footprint.sh

```diff
@@ -0,0 +1,556 @@
+
+###
+# which RUNTIME we use is picked up from the env in
+# common.bash. You can over-ride by setting RUNTIME in your env
+
+###
+# Define the cutoff checks for when we stop running the test
+  # Run up to this many containers
+NUM_CONTAINERS="${NUM_CONTAINERS:-200}"
+  # Run until we have consumed this much memory (from MemFree)
+MAX_MEMORY_CONSUMED="${MAX_MEMORY_CONSUMED:-256*1024*1024*1024}"
+  # Run until we have this much MemFree left
+MIN_MEMORY_FREE="${MIN_MEMORY_FREE:-2*1024*1024*1024}"
+
+# These paths come from the lib common.bash init sequence
+#PROXY_PATH
+#SHIM_PATH
+#HYPERVISOR_PATH
+
+# We monitor dockerd as we know it can grow as we run containers
+DOCKERD_PATH="${DOCKERD_PATH:-/usr/bin/dockerd}"
+
+# Tools we need to have installed in order to operate
+REQUIRED_COMMANDS="smem awk"
+
+# If we 'dump' the system caches before we measure then we get less
+# noise in the results - they show more what our un-reclaimable footprint is
+DUMP_CACHES="${DUMP_CACHES:-1}"
+
+# Affects the name of the file to store the results in
+TEST_NAME="${TEST_NAME:-fast-footprint-${PAYLOAD}}"
+
+############# end of configurable items ###################
+
+# vars to remember where we started so we can calc diffs
+base_mem_avail=0
+base_mem_free=0
+
+# dump the kernel caches, so we get a more precise (or just different)
+# view of what our footprint really is.
+function dump_caches() {
+	sudo bash -c "echo 3 > /proc/sys/vm/drop_caches"
+}
+
+function init() {
+	check_cmds $REQUIRED_COMMANDS
+	check_images "$PAYLOAD"
+
+	# Modify the test name if running with KSM enabled
+	check_for_ksm
+
+	# Use the common init func to get to a known state
+	init_env
+
+	# Prepare to start storing results
+	metrics_json_init
+
+	# Store up baseline measures
+	base_mem_avail=$(free -b | head -2 | tail -1 | awk '{print $7}')
+	base_mem_free=$(get_memfree)
+
+	# Store our configuration for this run
+	save_config
+}
+
+save_config(){
+	metrics_json_start_array
+
+	local json="$(cat << EOF
+	{
+		"testname": "${TEST_NAME}",
+		"payload": "${PAYLOAD}",
+		"payload_args": "${PAYLOAD_ARGS}",
+		"payload_runtime_args": "${PAYLOAD_RUNTIME_ARGS}",
+		"payload_sleep": ${PAYLOAD_SLEEP},
+		"ksm_settle_time": ${KSM_SETTLE_TIME},
+		"num_containers": ${NUM_CONTAINERS},
+		"parallelism": ${PARELLELISM},
+		"max_memory_consumed": "${MAX_MEMORY_CONSUMED}",
+		"min_memory_free": "${MIN_MEMORY_FREE}",
+		"dockerd_path": "${DOCKERD_PATH}",
+		"dump_caches": "${DUMP_CACHES}"
+	}
+EOF
+)"
+	metrics_json_add_array_element "$json"
+	metrics_json_end_array "Config"
+}
+
+function cleanup() {
+	# Finish storing the results
+	metrics_json_save
+
+	# This is what we have for now from the generics.
+	# Expect this to be improved (the generics) soon...
+
+	docker kill $(docker ps -qa)
+
+#	kill_processes_before_start
+}
+
+# See if KSM is enabled.
+# If so, ammend the test name to reflect that
+# FIXME - needs refactoring - see https://github.com/kata-containers/tests/issues/519
```

Comment:
Issue closed so can this comment be dropped?

Reply:
heh, that is the one bit of ksm code that did not get pulled out into the lib afaict. Let me do that now then...